packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -270,6 +275,18 @@ export class PendingStateManager implements IDisposable {
 
 		const messageContent = buildPendingMessageContent(message);
 
+		// Client Sequence Number should match per batch
+		// The server sequences messages in the order received, just like the pending message queue
+		if (this.pendingBatchBeginMessage?.clientSequenceNumber !== pendingMessage.batchStartCsn) {
+			this.logger?.sendErrorEvent({
+				eventName: "PendingClientSequenceNumberMismatch",
```

Comment:
Looks like some test is hitting this error

Reply:
Yeah had 2 bugs. One around grouped batching and the other around single-message batches of any kind.

I'm going to have to remove this for now, the fix for the grouped batching one deserves its own PR (and adding this check is not a priority - it was a learning opportunity for me)

Reply:
FYI @dannimad